### DISCUSSION TITLE
Redis client - RESP3 protocol

### DISCUSSION THREAD

**Discussion by uasan** [uNetworking/uWebSockets.js#259] [Q&A] [answered]
Hi.
Most of servers WebSocket need Redis clients for scaling.

Next major version Redis, implemented new version protocol - RESP3, there are two important innovations
1. [pubsub - do not block connections](https://github.com/antirez/RESP3/blob/master/spec.md#push-type)
2. [client side caching](http://antirez.com/news/130)

Current Redis clients for Node.js, are not very productive and do not support the new protocol.
Therefore, we plan to develop a new client for Redis.

Have you had any plans to implement your client for Redis?

Your Redis client (`uRedisClient`), could integrate `pubsub` functional with your `uWebSockets` server at a low level, this is a very demanding feature for your clients.

Thanks.

**Comment by ghost** [comment, marked as answer]
Redis is not an internet standard and not something facing web browsers. This project is about standard web IO, for the web.

I can't say for sure I'll never implement a Redis client but it'll not be part of this project anyways
